backend/src/jwt/decode.spec.js

```diff
@@ -0,0 +1,104 @@
+import Factory from '../seed/factories/index'
+import { getDriver } from '../bootstrap/neo4j'
+import decode from './decode'
+
+const factory = Factory()
+const driver = getDriver()
+
+// here is the decoded JWT token:
+// {
+//   role: 'user',
+//   locationName: null,
+//   name: 'Jenny Rostock',
+//   about: null,
+//   avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/sasha_shestakov/128.jpg',
+//   id: 'u3',
+//   email: 'user@example.org',
+//   slug: 'jenny-rostock',
+//   iat: 1550846680,
+//   exp: 1637246680,
+//   aud: 'http://localhost:3000',
+//   iss: 'http://localhost:4000',
+//   sub: 'u3'
+// }
+export const validAuthorizationHeader =
+  'Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJyb2xlIjoidXNlciIsImxvY2F0aW9uTmFtZSI6bnVsbCwibmFtZSI6Ikplbm55IFJvc3RvY2siLCJhYm91dCI6bnVsbCwiYXZhdGFyIjoiaHR0cHM6Ly9zMy5hbWF6b25hd3MuY29tL3VpZmFjZXMvZmFjZXMvdHdpdHRlci9zYXNoYV9zaGVzdGFrb3YvMTI4LmpwZyIsImlkIjoidTMiLCJlbWFpbCI6InVzZXJAZXhhbXBsZS5vcmciLCJzbHVnIjoiamVubnktcm9zdG9jayIsImlhdCI6MTU1MDg0NjY4MCwiZXhwIjoxNjM3MjQ2NjgwLCJhdWQiOiJodHRwOi8vbG9jYWxob3N0OjMwMDAiLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjQwMDAiLCJzdWIiOiJ1MyJ9.eZ_mVKas4Wzoc_JrQTEWXyRn7eY64cdIg4vqQ-F_7Jc'
+
+afterEach(async () => {
+  await factory.cleanDatabase()
+})
+
+describe('decode', () => {
+  let authorizationHeader
+  const returnsNull = async () => {
+    await expect(decode(driver, authorizationHeader)).resolves.toBeNull()
+  }
+
+  describe('given `null` as JWT Bearer token', () => {
+    beforeEach(() => {
+      authorizationHeader = null
+    })
+    it('returns null', returnsNull)
+  })
+
+  describe('given no JWT Bearer token', () => {
+    beforeEach(() => {
+      authorizationHeader = undefined
+    })
+    it('returns null', returnsNull)
+  })
+
+  describe('given malformed JWT Bearer token', () => {
+    beforeEach(() => {
+      authorizationHeader = 'blah'
+    })
+    it('returns null', returnsNull)
+  })
+
+  describe('given valid JWT Bearer token', () => {
+    beforeEach(() => {
+      authorizationHeader = validAuthorizationHeader
+    })
+    it('returns null', returnsNull)
```

Comment:
Why should this be null if the `authorizationHeader` is valid?

Reply:
Because the user is missing in the database. Real world example: You have a JWT cached in your browser and your user was deleted/disabled in the meantime.

Reply:
Possible! Just a matter of testing style. If the `describe` block would not make a change to the test setup (aka no `beforeEach` I usually omit it which sometimes leads to a cliffhanger like in this example here.

Reply:
_cliffhanger_, haha, I love it :smile: